src/infrastructure/validation/testing/TestHarness.java

```diff
@@ -390,5 +390,59 @@ public void runByName(String testNamePath){
       "Total Number of Failed Tests: "+Integer.toString(failedNumberOfTests)
     );
   }
+  
+  
+ /**
+  * Void main method to drive the test harness
+  * The args in main method must follow the below pattern-
+  * To run tests by category>> runByCategory categoryName
+  * To run tests by priority>> runByPriority priorityLevel
+  * To run all tests>> runAll
+  * To run a single test by its name>> runByName pathToTheTestStartingFrom-/src
+  */
+  public static void main(String args[]){
+    
+    ILogger logger = LoggerFactory.getLoggerInstance();
+    try{
+      if(args[0].equals("runByCategory")){
+        TestHarness myTestHarness = new TestHarness();  
+        myTestHarness.runByCategory(args[1]);
+      } 
+      else if(args[0].equals("runByPriority")){
+        int priority = Integer.parseInt(args[1]);
+        if(priority<0||priority>2) {
+          throw new IllegalArgumentException("Illegal priority level!");
+	      }
+        testHarness = new TestHarness();  
+        myTestHarness.runByPriority(priority);
```

Comment:
No class name specified. Follow the above suggested way of instantiating outside the try block

Reply:
Why was this resolved?